### ISSUE TITLE
Issues with Java files

### ISSUE THREAD

**Issue by chookity-pokk** [rswgnu/hyperbole#319]
So this was a bit of a hard issue to pin to Hyperbole, so I am going to try and be as explicit as possible in explaining my issue so this might be a longer read.

Essentially is seems like with large Java files (500+ lines), my Emacs would freeze and max out a CPU core. I tried using the built in profiler but it always seems to give a similar output to what is below, but change `newline` for any other built in function. At first I thought it was an issue with Emacs itself, because this was reproducible 100% of the time. Though when dissecting my [init](https://gitlab.com/greenbeast/dotfiles/-/blob/main/init.org) file, I found that it only happens if I have `(hyperbole-mode 1)`.

I've reproduced this with Hyperbole a few dozen times and also had it not happen when I comment out the above snippet from my init. Now the hard part to pin this on hyperbole is that I've had a few rare occasions where I have hyperbole active, but it takes a few minutes before my Emacs freezes. Though, about 90% of the time this happens within the time it takes to write a single function. 

This very well might not be a Hyperbole issue, as the logs never mention Hyperbole. Though in my testing it seems like Hyperbole is at least playing some kind of a role in this.

Steps to reproduce:
Open a java file (500+ lines)
Scroll quickly (CTRL+ Up/Down Arrow)
After scrolling through a couple times it almost always freezes
If not just try writing or deleting any text for a bit and it freezes

```
       11049  61% - command-execute
       11049  61%  - call-interactively
       11046  61%   - funcall-interactively
        9983  55%    - newline
        9983  55%     - self-insert-command
        9983  55%      - c-before-change
        9983  55%       - c-laomib-invalidate-cache
        9983  55%        - c-parse-state
        9983  55%         - c-parse-state-1
        9983  55%          - c-parse-state-get-strategy
        9877  55%           - c-get-fallback-scan-pos
        8735  48%            - beginning-of-defun
        8278  46%             - beginning-of-defun-raw
        6163  34%              - syntax-ppss
          16   0%                 #<compiled -0x42ef295dad943a2>
```

**Comment by rswgnu** [comment]
Hi Hank:

You are running one of the newer pre-releases of Hyperbole, right?  If not, switch to that and see if this goes away.  We are readying a new major release and would certainly be happy to resolve any issue like this if related to Hyperbole but would need more of a pointer, e.g. a stack trace showing a Hyperbole call in the mix to look at it.

**Comment by chookity-pokk** [comment]
So it seems like my hyperbole was downloading but not updating (via quelpa). So I thought I was up to date on the package but it turns out I wasn't. Actually updating hyperbole fixed the issue.

**Comment by rswgnu** [comment]
Thanks, Hank.  Lots of good things to check out in {C-h h d n}.-- rswOn May 12, 2023, at 12:09 PM, Hank Greenburg ***@***.***> wrote:﻿
Closed #319 as completed.
